app/src/main/java/io/gnosis/safe/notifications/NotificationRepository.kt

```diff
@@ -224,9 +228,17 @@ class NotificationRepository(
             }
             .onFailure {
                 registrationUpdateFailed = true
+                resetFirebaseToken()
```

Comment:
This will loop, if the new token doesn't fix the error. For example if we try to register an a key that's not an owner.